music/src/core/visualizer.ts

```diff
@@ -163,22 +162,20 @@ export abstract class BaseVisualizer {
 
   protected scrollIntoViewIfNeeded(
       scrollIntoView: boolean, activeNotePosition: number) {
-    if (scrollIntoView && this.parentElement) {
-      // See if we need to scroll the container.
+    if (scrollIntoView) {
       const containerWidth = this.parentElement.getBoundingClientRect().width;
-      if (activeNotePosition >
-          (this.parentElement.scrollLeft + containerWidth)) {
-        this.parentElement.scrollLeft = activeNotePosition - 20;
-      }
+      this.parentElement.scrollLeft = activeNotePosition - containerWidth * 0.5;
     }
   }
 
   protected getNoteStartTime(note: NoteSequence.INote) {
-    return this.sequenceIsQuantized ? note.quantizedStartStep : note.startTime;
+    return this.sequenceIsQuantized ? note.quantizedStartStep : 
+        Math.round(note.startTime * 100000000) / 100000000;
```

Comment:
Is this the same as `note.startTime.toFixed(...)` ? Either way I think this is a bit too much math to do in a loop over and over again for every note. What if you just add and subtract a small epsilon from `this.getNoteStartTime(playedNote)`, if the problem is just a floating point error? That way it's only done once per redraw.

Reply:
I tried other approaches, but they weren't that easy because you never know if epsilon must be positive or negative, therefore the need for Math.round() which is pretty efficient, and the extra * and / are very fast as well (they are bare processor primitives). I've tested it in very heavy load contexts without a problem at all.

Reply:
Moreover, toFixed() returns a String, not a Number.

Reply:
Tried to find an [easier solution](https://stackoverflow.com/questions/1458633/how-to-deal-with-floating-point-number-precision-in-javascript), but they involved more maths or string parsing. I propose leaving it as is and evolve later if some other solution raises.

Reply:
@notwaldorf I've done some extra-latency testings with solid and consistent results:

```typescript
  protected getNoteStartTime(note: NoteSequence.INote) {
    return this.sequenceIsQuantized ? note.quantizedStartStep : 
        Math.round(note.startTime * 100000000) / 100000000;
  }

  protected getNoteStartTimeOld(note: NoteSequence.INote) {
    return this.sequenceIsQuantized ? note.quantizedStartStep : note.startTime;
  }

  protected testRounding(n: number): number {
    // Data setup
    let notes = new Array<NoteSequence.INote>(n);
    for (let i = 0; i < n; ++i) {
      const note = new NoteSequence.Note;
      note.startTime = 1/i;
      notes[i] = note;
    }

    // First roundup
    let timeOldBegin = (new Date()).getTime();
    for (let i = 0; i < n; ++i) { this.getNoteStartTimeOld(notes[i]); }
    let timeOldEnd = (new Date()).getTime();
    let timeNewBegin = (new Date()).getTime();
    for (let i = 0; i < n; ++i) { this.getNoteStartTime(notes[i]); }
    let timeNewEnd = (new Date()).getTime();

    // Second roundup to avoid data caching effect
    timeOldBegin = (new Date()).getTime();
    for (let i = 0; i < n; ++i) { this.getNoteStartTimeOld(notes[i]); }
    timeOldEnd = (new Date()).getTime();
    timeNewBegin = (new Date()).getTime();
    for (let i = 0; i < n; ++i) { this.getNoteStartTime(notes[i]); }
    timeNewEnd = (new Date()).getTime();
    
    console.log('oldPost - oldPre  = ', timeOldEnd - timeOldBegin);
    console.log('newPost - newPre  = ', timeNewEnd - timeNewBegin);
    console.log('Latency per request = ', 
      ((timeNewEnd-timeNewBegin) - (timeOldEnd - timeOldBegin)) / n);
    return (timeNewEnd-timeNewBegin) - (timeOldEnd - timeOldBegin);
  }
```
Running it in Chrome console over an instance having `this.sequenceIsQuantized == null`(which is where the new quantization applies) yields following results:
```console
> this.testRounding(10000000);
oldPost - oldPre  =  477
newPost - newPre  =  525
Latency per request =  0.0000048
48
>
```
I guess five millionth of millisecond (4.8 nanosecond) per call is good enough latency to let it go. Please trust in wonderful V8 JavaScript JIT compiler magic and accept the PR :-).

Reply:
sg, no worries